src/docfx/lib/log/Telemetry.cs

```diff
@@ -64,9 +64,9 @@ public static IDisposable TrackingOperationTime(TelemetryName name)
             return new PerfScope(name.ToString());
         }
 
-        public static void TrackErrorCount(string code, ErrorLevel level)
+        public static void TrackErrorCount(string code, ErrorLevel level, string? name)
         {
-            s_errorCountMetric.TrackValue(1, code, level.ToString(), "User", s_os, s_version, s_repo, s_branch, s_correlationId);
+            s_errorCountMetric.TrackValue(1, code, level.ToString(), CoalesceEmpty(name), "User", s_os, s_version, s_repo, s_branch, s_correlationId);
```

Comment:
how about use Dictionary<string, string> properties, then the caller can track any dimensions they want.

use s_telemetryClient.TrackMetrics()

Reply:
Unfortunately, the dimension count for metric is fixed. (It is reasonable because pre-aggregation needs to accumulate values based on a fixed dimension).